File: java-samples/corda5-negotiation-cordapp/workflows/src/main/java/com/r3/developers/samples/negotiation/workflows/util/ListProposalArgs.java
package com.r3.developers.samples.negotiation.workflows.util;

import java.util.UUID;

public class ListProposalArgs {

    private UUID proposalID;
    private int amount;
    private String buyer;
    private String seller;
    private String proposer;
    private String proposee;

    public ListProposalArgs() {
    }

    public ListProposalArgs(UUID proposalID, int amount, String buyer, String seller, String proposer, String proposee) {
        this.proposalID = proposalID;
        this.amount = amount;
        this.buyer = buyer;
        this.seller = seller;
        this.proposer = proposer;
        this.proposee = proposee;
    }

    public UUID getProposalID() {
        return proposalID;
    }

    public int getAmount() {
        return amount;
    }

    public String getBuyer() {
        return buyer;
    }

    public String getSeller() {
        return seller;
    }

    public String getProposer() {
        return proposer;
    }

    public String getProposee() {
        return proposee;
    }


}

File: kotlin-samples/corda5-negotiation/workflows/src/main/kotlin/com/r3/developers/samples/negotiation/workflows/util/ListProposalArgs.kt
package com.r3.developers.samples.negotiation.workflows.util

import java.util.*

class ListProposalArgs {
    var proposalID: UUID? = null
        private set
    var amount = 0
        private set
    var buyer: String? = null
        private set
    var seller: String? = null
        private set
    var proposer: String? = null
        private set
    var proposee: String? = null
        private set

    constructor()
    constructor(proposalID: UUID?, amount: Int, buyer: String?, seller: String?, proposer: String?, proposee: String?) {
        this.proposalID = proposalID
        this.amount = amount
        this.buyer = buyer
        this.seller = seller
        this.proposer = proposer
        this.proposee = proposee
    }
}
